app/components/editor/layouts/BaseLayout.ts

```diff
@@ -32,6 +37,37 @@ export default class BaseLayout extends TsxComponent<LayoutProps> {
     window.removeEventListener('resize', () => this.props.windowResizeHandler(this.mins));
   }
 
+  async setMins(
+    restSlots: ILayoutSlotArray,
+    bar1Slots: ILayoutSlotArray,
+    bar2Slots?: ILayoutSlotArray,
+  ) {
+    const rest = await this.calculateMinimum(restSlots);
+    const bar1 = await this.calculateMinimum(bar1Slots);
+    const bar2 = await this.calculateMinimum(bar2Slots);
+    this.mins = { rest, bar1, bar2 };
+  }
+
+  async minsFromSlot(slot: LayoutSlot) {
+    await this.$nextTick();
```

Comment:
this single line makes multiple methods here `async` and adds multiple `this.$nextTick()` calls to the `setMins` method. Can we avoid this line?

Reply:
that's the core tradeoff of this refactor-- if we want to access elements sizes from the element itself we need to wait a render cycle before being able to access the `componentInstance` of those components